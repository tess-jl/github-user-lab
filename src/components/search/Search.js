import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { fetchUserAction } from '../../actions/userInfoActions';
import { fetchUserRepos } from '../../services/gitHubApi';


const Search = () => {
  const [usernameToSearch, setUsernameToSearch] = useState('');
  const dispatch = useDispatch();

  const handleSubmit = () => {
    event.preventDefault();
    dispatch(fetchUserAction(usernameToSearch));
    dispatch(fetchUserRepos(usernameToSearch));
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={usernameToSearch} placeholder="GitHub User" onChange={({ target }) => setUsernameToSearch(target.value) }/> 
      <button>search</button>
    </form>
  );
};

export default Search; 
