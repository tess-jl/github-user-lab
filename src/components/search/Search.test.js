import React from 'react';
import { shallow } from 'enzyme';
import Search from './Search';
import { Provider } from 'react-redux';
import store from '../../store';

describe('Search component', () => {
  it('renders Search', () => {
    const wrapper = shallow(<Provider store={store}><Search /></Provider>).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
