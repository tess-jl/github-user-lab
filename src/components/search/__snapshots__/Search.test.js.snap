// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders Search 1`] = `<Search />`;
